components/locid_transform/src/canonicalizer.rs

```diff
@@ -230,25 +224,25 @@ impl LocaleCanonicalizer {
     #[doc = icu_provider::gen_any_buffer_unstable_docs!(ANY, Self::new)]
     pub fn try_new_with_any_provider(
         provider: &(impl AnyProvider + ?Sized),
-    ) -> Result<LocaleCanonicalizer, LocaleTransformError> {
+    ) -> Result<Self, LocaleTransformError> {
         let expander = LocaleExpander::try_new_with_any_provider(provider)?;
-        Self::try_new_with_expander_unstable(&provider.as_downcasting(), expander)
+        Self::try_new_with_expander_compat(&provider.as_downcasting(), expander)
     }
 
     // Note: This is a custom impl because the bounds on LocaleExpander::try_new_unstable changed
     #[doc = icu_provider::gen_any_buffer_unstable_docs!(BUFFER, Self::new)]
     #[cfg(feature = "serde")]
     pub fn try_new_with_buffer_provider(
         provider: &(impl BufferProvider + ?Sized),
-    ) -> Result<LocaleCanonicalizer, LocaleTransformError> {
+    ) -> Result<Self, LocaleTransformError> {
         let expander = LocaleExpander::try_new_with_buffer_provider(provider)?;
-        Self::try_new_with_expander_unstable(&provider.as_deserializing(), expander)
+        Self::try_new_with_expander_compat(&provider.as_deserializing(), expander)
```

Comment:
Please change this to call `try_new_with_expander_with_buffer_provider` instead.

Reply:
This fails bound check:
```
239 |         Self::try_new_with_expander_with_buffer_provider(&provider.as_deserializing(), expander)
    |         ------------------------------------------------ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `icu_provider::BufferProvider` is not implemented for `DeserializingBufferProvider<'_, impl BufferProvider + ?Sized>`
    |         |
    |         required by a bound introduced by this call
```

Is it why you asked me to unpack the macro above?

Reply:
Take out the `.as_deserializing()` when you call `_with_buffer_provider`